packages/jazz-webhook/src/webhook.ts

```diff
@@ -0,0 +1,373 @@
+    }
+
+    this.state.$jazz.delete(webhookId);
+    webhook.$jazz.set("active", false);
+  }
+
+  /**
+   * Starts monitoring a CoValue and emitting webhooks when it changes.
+   *
+   * @param webhookId - The ID of the webhook to start monitoring
+   */
+  private subscribe(webhook: WebhookRegistration) {
+    const emitter = new WebhookEmitter(webhook, this.options);
+
+    this.activeSubscriptions.set(webhook.$jazz.id, () => {
+      emitter.stop();
+      this.activeSubscriptions.delete(webhook.$jazz.id);
+    });
+  }
+
+  /**
+   * Starts monitoring all active webhooks in the registry.
+   *
+   * This method iterates through all webhooks in the registry and starts
+   * subscriptions for any active webhooks that don't already have active
+   * subscriptions. This is useful for resuming webhook monitoring after
+   * a shutdown or when loading an existing registry.
+   */
+  async start(): Promise<void> {
+    if (this.rootUnsubscribe) {
+      return;
+    }
+
+    // TODO: this would be much more efficient with subscription diffs
+    const createAndDeleteSubscriptions = (
+      registry: co.loaded<typeof RegistryState, { $each: true }>,
+    ) => {
+      for (const webhook of Object.values(registry)) {
+        const exists = this.activeSubscriptions.has(webhook.$jazz.id);
+        if (webhook.active && !exists) {
+          this.subscribe(webhook);
+        }
+      }
+      for (const [id, unsubscribe] of this.activeSubscriptions.entries()) {
+        if (!registry[id]) {
+          unsubscribe();
+          this.activeSubscriptions.delete(id);
+        }
+      }
+    };
+
+    this.rootUnsubscribe = this.state.$jazz.subscribe(
+      {
+        resolve: {
+          $each: true,
+        },
+      },
+      createAndDeleteSubscriptions,
+    );
+  }
+
+  shutdown(): void {
+    for (const unsubscribe of this.activeSubscriptions.values()) {
+      unsubscribe();
+    }
+    this.rootUnsubscribe?.();
+    this.rootUnsubscribe = null;
+  }
+}
+
+class WebhookError extends Error {
+  constructor(
+    message: string,
+    public readonly retryAfterMs?: number,
+  ) {
+    super(message);
+  }
+}
+
+/**
+ * Manages webhook emission with queuing, retry logic, and exponential backoff.
+ *
+ * The WebhookEmitter handles sending HTTP POST requests to webhook callbacks when
+ * a CoValue changes. It implements several key behaviors:
+ *
+ */
+class WebhookEmitter {
+  successMap: Promise<co.loaded<typeof SuccessMap, { $each: true }>>;
+  loadedSuccessMap: co.loaded<typeof SuccessMap, { $each: true }> | undefined;
+  pending = new Map<
+    string, // stringified TransactionID
+    { nRetries: number; timeout?: NodeJS.Timeout }
+  >();
+  unsubscribe: () => void;
+
+  constructor(
+    private webhook: WebhookRegistration,
+    private options: JazzWebhookOptions = {},
+  ) {
+    this.successMap = webhook.$jazz
+      .ensureLoaded({ resolve: { successMap: { $each: true } } })
```

Comment:
We actually don't need `$each` here because `successMap` doesn't have child coValues (the value is just a boolean)

Typescript should've catched this, it's probably a bug in our resolve